RedHat-UX/red-hat-design-system
# docs(feedback): added related items to docs

## What I did

1. Added a CSV file to store groups of related elements and patterns. Each row is a group of comma-separated elements/patterns.
2. Added `relatedItems` filter and `getLinkInfoForItems` function to `rhds.cjs`, which does the following:
    a. Accepts a string of comma-separated pattern names and/or element tagNames.
    b. Provides an array of objects with link text and url for each item.
    c. Returns a maximum of the first three and sorts those items alphabetically by link text.
4. Added logic to `feedback.html` to get related items for patterns and elements.
5. On several elements and patterns, set a variable with a comma-separated list of related elements and patterns.
6. Added list styles for related items lists.


## Testing Instructions

1. On the [All elements](https://deploy-preview-942--red-hat-design-system.netlify.app/elements/) page, verify that the **Foundations** section appears instead of the **Related patterns or components** section.
2. Check the links for the **Related patterns or components** section of each element page. _(Use the [yaml file](https://github.com/RedHat-UX/red-hat-design-system/pull/942/files#diff-de80a32ed117ccaab6f0426d14d5a3b053bc830a358ac45b10b6e4885931e9f5) to verify accuracy.)_
3. On the [All patterns](https://deploy-preview-942--red-hat-design-system.netlify.app/patterns/) page, verify that the **Foundations** section appears instead of the **Related patterns or components** section.
4. Check the links for the **Related patterns or components** section of each patten page. _(Use the [yaml file](https://github.com/RedHat-UX/red-hat-design-system/pull/942/files#diff-de80a32ed117ccaab6f0426d14d5a3b053bc830a358ac45b10b6e4885931e9f5) to verify accuracy.)_

## Notes to Reviewers
As @coreyvickery @zeroedin discussed, we decided to use front matter after all since the related items section for a component appears on every single page of that component. Instead we'll use this [yaml file](https://github.com/RedHat-UX/red-hat-design-system/pull/942/files#diff-de80a32ed117ccaab6f0426d14d5a3b053bc830a358ac45b10b6e4885931e9f5).  

The yaml file is structured like this:
```
[element/pattern whose docs pages have related items at the bottom]:
 - [related item]
 - [related item]
 - [related item]
 - [ignored item]
```

So given the following example CSV:
```
rh-badge:
 - rh-button
 - rh-avatar
 - rh-tag
rh-tooltip:
 - rh-badge
 - rh-button
 - rh-popover
 - rh-avatar
 - promo
```

- The docs page for Badge would list Avatar, Button, and Tag (all automatically alphabetized, btw)
- The docs page for Tooltip would only have Badge, Button, and Popover because it ignores anything after the third related item. (But you could leave extra ones them there in case we ever want a record of them or delete one of the first three.)
- The docs page for Button would not have related items. You'd need to add its own line `rh-button, rh-avatar, rh-tag, rh-badge` to do that. (This addresses @coreyvickery's  interest in having the related items vary from component to component.)
